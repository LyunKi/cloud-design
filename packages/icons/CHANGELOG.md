# @cloud-design/icons

## 0.2.1

### Patch Changes

- Fix: handle illegal icon name.

## 0.2.0

### Minor Changes

- 增加 Text 实现

## 0.1.0

### Minor Changes

- init cloud design components
