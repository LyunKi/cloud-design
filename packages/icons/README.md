---
title: Icon
nav:
  title: Components
  path: /components
group:
  title: Icon
  path: /icon
  order: 2
toc: menu
---

# `@cloud-design/icons`

cloud design icons for react-native

## Install

```sh
$ npm install @cloud-design/icons
```

## Usage

<code src="./demo/index.tsx"></code>
