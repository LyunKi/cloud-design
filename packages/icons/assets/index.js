const fs = require('fs')

const dir = fs.readdirSync('./fill')

const iconNames = dir.map((f) => f.split('.')[0])

console.log(iconNames.join(','))
