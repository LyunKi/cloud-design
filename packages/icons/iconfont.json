{
  "symbol_url": "请参考README.md，复制官网提供的JS链接。如果提供了local_svgs，则url选填",
  "use_typescript": false,
  "save_dir": "./src/iconfont",
  "trim_icon_prefix": "eva",
  "default_icon_size": 18,
  "local_svgs": ""
}
