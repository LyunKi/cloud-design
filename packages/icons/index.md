---
title: Icon
nav:
  title: Components
  path: /components
group:
  title: Icon
  path: /icon
  order: 2
toc: menu
---

# `@cloud-design/icons`

react-native icon 组件

## 安装

```sh
$ npm install @cloud-design/icons
```

<API src='./src/index.tsx'></API>

## icon 全览

<code src="./demo/index.tsx"></code>
