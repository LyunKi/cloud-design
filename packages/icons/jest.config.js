module.exports = {
  ...require('../../jest.config.base'),
  preset: 'react-native',
  testEnvironment: 'jsdom',
  rootDir: '../..',
  roots: [`<rootDir>/packages/icons`],
  testMatch: ['<rootDir>/packages/icons/test/**/*.[jt]s?(x)'],
}
