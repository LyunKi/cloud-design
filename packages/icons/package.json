{
  "name": "@cloud-design/icons",
  "description": "React Native icons",
  "module": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "generate": "pnpm exec iconfont-rn",
    "compile": "tsc"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "react-native": "^0.69.3"
  },
  "dependencies": {
    "react-native-svg": "^12.4.3"
  },
  "devDependencies": {
    "@cloud-dragon/tsconfig-react": "^0.0.1",
    "@types/react": "^18.0.15",
    "@types/react-native": "^0.69.5",
    "react": "^18.2.0",
    "react-native-iconfont-cli": "^2.2.4"
  }
}
