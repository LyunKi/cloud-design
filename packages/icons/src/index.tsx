import React, { Ref } from 'react'
import { Animated } from 'react-native'
import { AnimationConfig } from './animation/animation'
import {
  getIconAnimation,
  SupportedAnimation,
} from './animation/animationRegistry'
import { findIconByName } from './generated'

export type IconAnimationProp = {
  type: SupportedAnimation
  config?: AnimationConfig
}

export interface IconProps {
  name: string
  size?: number
  color?: string
  animation?: IconAnimationProp
}

export interface IconRef {
  startAnimation?: Function
  stopAnimation?: Function
}

export const Icon = React.forwardRef(function (
  props: IconProps,
  ref: Ref<IconRef>
) {
  const { name, size = 24, color, animation } = props
  const Icon = findIconByName(name)
  const animationInstance = React.useMemo(() => {
    if (!animation) {
      return
    }
    const { type, config } = animation
    return getIconAnimation(type, config)
  }, [animation])
  React.useImperativeHandle(ref, () => {
    return {
      startAnimation: animationInstance?.start,
      stopAnimation: animationInstance?.stop,
    }
  })
  return Icon ? (
    <Animated.View {...animationInstance?.toProps}>
      {React.createElement(Icon, {
        width: size,
        height: size,
        fill: color,
      })}
    </Animated.View>
  ) : null
})
