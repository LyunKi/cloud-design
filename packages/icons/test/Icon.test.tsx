import { render } from '@testing-library/react-native'
import { Icon } from '@cloud-design/icons'
import React from 'react'

describe('<Icon />', () => {
  it('render correctly', () => {
    const snapshot = render(
      <Icon
        size={24}
        color="red"
        animation={{ type: 'zoom' }}
        name="facebook"
      />
    ).toJSON()
    expect(snapshot).toMatchSnapshot()
  })
})
