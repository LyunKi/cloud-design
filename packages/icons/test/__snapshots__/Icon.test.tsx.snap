// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> render correctly 1`] = `
<View
  collapsable={false}
  style={Object {}}
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={24}
    bbWidth={24}
    fill="red"
    focusable={false}
    height={24}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width={24}
  >
    <RNSVGGroup
      fill={4294901760}
      propList={
        Array [
          "fill",
        ]
      }
    >
      <RNSVGGroup>
        <RNSVGGroup>
          <RNSVGRect
            height="24"
            matrix={
              Array [
                -1,
                1.2246467991473532e-16,
                -1.2246467991473532e-16,
                -1,
                24,
                24,
              ]
            }
            opacity={0}
            width="24"
            x={0}
            y={0}
          />
          <RNSVGPath
            d="M17 3.5a.5.5 0 0 0-.5-.5H14a4.77 4.77 0 0 0-5 4.5v2.7H6.5a.5.5 0 0 0-.5.5v2.6a.5.5 0 0 0 .5.5H9v6.7a.5.5 0 0 0 .5.5h3a.5.5 0 0 0 .5-.5v-6.7h2.62a.5.5 0 0 0 .49-.37l.72-2.6a.5.5 0 0 0-.48-.63H13V7.5a1 1 0 0 1 1-.9h2.5a.5.5 0 0 0 .5-.5z"
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
